sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/DatetimeShardingAlgorithm.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.core.strategy.algorithm.sharding;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.shardingsphere.api.sharding.standard.PreciseShardingValue;
+import org.apache.shardingsphere.api.sharding.standard.RangeShardingValue;
+import org.apache.shardingsphere.api.sharding.standard.StandardShardingAlgorithm;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.LinkedHashSet;
+import java.util.Properties;
+
+/**
+ * Datetime sharding algorithm.
+ * 
+ * <p>Shard by `y = floor(x/v)` algorithm, which means y begins from 0.
+ * v is `PARTITION_VOLUME`, and the minimum time unit is 1 sec.
+ * `SINCE_DATETIME` decides the beginning datetime to shard. </p>
+ */
+public final class DatetimeShardingAlgorithm implements StandardShardingAlgorithm<Comparable<?>> {
+    
+    private static final String PARTITION_VOLUME = "partition.volume";
+    
+    private static final String SINCE_DATETIME = "since.datetime";
+    
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
`SimpleDateFormat` is not thread safe, is it possible to call `SimpleDateFormat.parse()` concurrently? if so, perhaps use `LocalDateTime` and `DateTimeFormatter` is better

Reply:
@ SanmuWangZJU

> `SimpleDateFormat` is not thread safe, is it possible to call `SimpleDateFormat.parse()` concurrently? if so, perhaps use `LocalDateTime` and `DateTimeFormatter` is better

Hi, Thanks for your suggestion which helps raize #5341, welcome your review and comments.